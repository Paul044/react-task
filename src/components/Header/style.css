.logo {
    color: #f55263;
    font-size: 20px;
    padding: 20px 100px 0 100px;
}

.header {
    background-image: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)), url(img/background.png);

    background-color: red;
}
.search-btn{
    border: none;
    background-color: #E9E8E8;
    color: #f55263;
    text-transform: uppercase;
    font-weight: bold;
    padding: 10px 30px;
    float: right;
    cursor: pointer;
    display: block;
}
.search-btn:hover {
    color: #E9E8E8;
    background-color: #f55263;
}