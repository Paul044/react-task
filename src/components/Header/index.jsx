import React from 'react';
import DetailPage from 'components/DetailPage/index';
import SearchForm from 'components/SearchForm/index';
import { Route, Switch } from 'react-router-dom';

import * as style from './style.css';

class Header extends React.Component {
    handleGoToSearch() {
        this.props.history.push('/');
    }

    render() {
        return (
            <div className={style.header}>
                <div className={style.logo}>
                    <span>netflixrullete</span>
                    <Route
                        path="/film"
                        render={() => (
                            <button
                                className={style.searchBtn}
                                onClick={this.handleGoToSearch.bind(this)}
                            >SEARCH
                            </button>
                        )}
                    />
                </div>

                <Route path="/film/:id" component={DetailPage} />
                <Switch>
                    <Route path="/" exact component={SearchForm} />
                    <Route path="/search/:query" component={SearchForm} />
                </Switch>
            </div>
        );
    }
}

export default Header;
