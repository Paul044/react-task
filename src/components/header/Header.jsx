import React from 'react';
import PropTypes from 'prop-types';
import DetailPage from 'components/detail-page/DetailPage';
import SearchForm from 'components/search-form/SearchForm';
import AdditionalInfo from 'components/additional-info/AdditionalInfo';
import { AppStates } from 'components/app/app.constants';
import * as s from './Header.css';

class Header extends React.Component {
    render() {
        return (
            <div className={s.header}>
                <div className={s.logo}>
                    <span>netflixrullete</span>
                    {this.props.appState === AppStates.DETAILED &&
                        <button
                            className={s.searchBtn}
                            onClick={this.props.goToSearchClick}
                        >SEARCH
                        </button>}
                </div>

                {this.props.show ?
                    <DetailPage
                        show={this.props.show}
                    /> :
                    <SearchForm
                        searchBtnClick={this.props.searchBtnClick}
                    />
                }
                <AdditionalInfo
                    appState={this.props.appState}
                    show={this.props.show}
                    items={this.props.items}
                    sortByDateClick={this.props.sortByDateClick}
                    sortRatingClick={this.props.sortRatingClick}
                />
            </div>
        );
    }
}

Header.propTypes = {
    sortByDateClick: PropTypes.func,
    sortRatingClick: PropTypes.func,
    appState: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.object),
    show: PropTypes.object,
    goToSearchClick: PropTypes.func,
    searchBtnClick: PropTypes.func,
};

Header.defaultProps = {
    sortByDateClick: () => {},
    sortRatingClick: () => {},
    appState: '',
    items: [],
    show: undefined,
    goToSearchClick: () => {},
    searchBtnClick: () => {},
};

export default Header;
