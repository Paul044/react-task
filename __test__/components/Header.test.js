import React from 'react';
import { shallow } from 'enzyme';
import Header from 'components/Header';


describe('<Header />', () => {
    const wrapper = shallow(<Header />);
    it('should be like snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
