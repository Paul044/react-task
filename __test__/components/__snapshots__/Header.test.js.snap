// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be like snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        >
          <span>
            netflixrullete
          </span>
          <Route
            path="/film"
            render={[Function]}
          />
        </div>,
        <Route
          component={[Function]}
          path="/film/:id"
        />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/search/:query"
          />
        </Switch>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              netflixrullete
            </span>,
            <Route
              path="/film"
              render={[Function]}
            />,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "netflixrullete",
            },
            "ref": null,
            "rendered": "netflixrullete",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/film",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/film/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/search/:query"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/search/:query",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            <span>
              netflixrullete
            </span>
            <Route
              path="/film"
              render={[Function]}
            />
          </div>,
          <Route
            component={[Function]}
            path="/film/:id"
          />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/search/:query"
            />
          </Switch>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                netflixrullete
              </span>,
              <Route
                path="/film"
                render={[Function]}
              />,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "netflixrullete",
              },
              "ref": null,
              "rendered": "netflixrullete",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/film",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/film/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/search/:query"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/search/:query",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
